instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go

```diff
@@ -209,8 +209,10 @@ func (c *config) handleRPC(ctx context.Context, rs stats.RPCStats, isServer bool
 		elapsedTime := float64(rs.EndTime.Sub(rs.BeginTime)) / float64(time.Millisecond)
 
 		c.rpcDuration.Record(wctx, elapsedTime, metric.WithAttributes(metricAttrs...))
-		c.rpcRequestsPerRPC.Record(wctx, atomic.LoadInt64(&gctx.messagesReceived), metric.WithAttributes(metricAttrs...))
-		c.rpcResponsesPerRPC.Record(wctx, atomic.LoadInt64(&gctx.messagesSent), metric.WithAttributes(metricAttrs...))
+		if gctx != nil {
```

Comment:
Can you add a comment here?  In what cases are we skipping the recording?

Reply:
Actually, I was trying to wrap `otelgrpc.StatsHandler` with `otelgrpc/filters` support in the company's internal code.

---

What comment is correct?

Reply:
I'm not sure... Why are you skipping these lines when the context is nil?  Any why skip `rpcRequestsPerRPC` and `pcResponsesPerRPC`, but not `rpcDuration`?

Reply:
We want to just use [filters.HealthCheck](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/v1.21.1/instrumentation/google.golang.org/grpc/otelgrpc/filters/filters.go#L156-L158), like https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4575.

Reply:
Well, already checks `gctx` nil check on https://github.com/open-telemetry/opentelemetry-go-contrib/blob/d50aa19a4d1d8971fa5796fdf01ed27ca5e8b090/instrumentation/google.golang.org/grpc/otelgrpc/stats_handler.go#L153, This change will be more safer (without wraps otelgrpc.StatsHandler)